src/app/services/user.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { User } from '../models/User';
+import { HttpClient } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+import { AuthService } from './auth/auth.service';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  constructor(private http: HttpClient, private authService: AuthService) { }
+
+  getAllUsers(page: number, size: number) : Observable<User[]>{
+    return this.http.get<User[]>(environment.userUri);
+  }
+
+  getUserById(id:number) : Observable<User>{
+    return this.http.get<User>(`${environment.userUri}/${id}`);
+  }
+
+  createOrUpdateUser(user: User, id:number) : Observable<User>{
+    return this.http.put<User>(`${environment.userUri}/${id}`, user);
+  }
+
+  updateUser(user: User, id: number) : Observable<User>{
+    return this.http.patch<User>(`${environment.userUri}/${id}`, user);
+  }
+
+  updateProfilePic(profilePic: FormData, id: number){
+    return this.http.patch<string>(`${environment.userProfile}/${id}`, profilePic);
```

Comment:
Make use of `${userUri}/picture` or similar rather than referencing userProfile (which doesn't exist in this pr)